mlflow/pytorch/_lightning_autolog.py

```diff
@@ -323,111 +324,31 @@ def __init__(
-            else:
-                checkpoint_model_filename = "latest_checkpoint.pth"
-            checkpoint_metrics_filename = "latest_checkpoint_metrics.json"
-            checkpoint_artifact_dir = "checkpoints"
-        else:
-            if self.save_freq == "epoch":
-                sub_dir_name = f"epoch_{current_epoch}"
-            else:
-                sub_dir_name = f"global_step_{trainer.global_step}"
-
-            if self.save_weights_only:
-                checkpoint_model_filename = "checkpoint.weights.pth"
-            else:
-                checkpoint_model_filename = "checkpoint.pth"
-            checkpoint_metrics_filename = "checkpoint_metrics.json"
-            checkpoint_artifact_dir = f"checkpoints/{sub_dir_name}"
-
-        self.client.set_tag(
-            self.run_id,
-            LATEST_CHECKPOINT_ARTIFACT_TAG_KEY,
-            f"{checkpoint_artifact_dir}/{checkpoint_model_filename}",
+        super().__init__(
+            client=client,
+            run_id=run_id,
+            monitor=monitor,
+            mode=mode,
+            save_best_only=save_best_only,
+            save_weights_only=save_weights_only,
+            save_freq=save_freq,
         )
-
-        self.client.log_dict(
-            self.run_id,
-            {**metric_dict, "epoch": current_epoch, "global_step": trainer.global_step},
-            f"{checkpoint_artifact_dir}/{checkpoint_metrics_filename}",
+        self.trainer = None
+
+    def save_checkpoint(self, filepath: str):
+        # Note: `trainer.save_checkpoint` implementation contains invocation of
+        # `self.strategy.barrier("Trainer.save_checkpoint")`,
+        # in DDP training, this callback is only invoked in rank 0 process,
+        # the `barrier` invocation causes deadlock,
+        # so I implement `save_checkpoint` instead of
+        # calling `trainer.save_checkpoint`.
+        checkpoint = self.trainer._checkpoint_connector.dump_checkpoint(
+            self.save_weights_only
         )
+        self.trainer.strategy.save_checkpoint(checkpoint, filepath)
 
-        tmp_dir = create_tmp_dir()
-        try:
-            tmp_model_save_path = os.path.join(tmp_dir, checkpoint_model_filename)
-            # Note: `trainer.save_checkpoint` implementation contains invocation of
-            # `self.strategy.barrier("Trainer.save_checkpoint")`,
-            # in DDP training, this callback is only invoked in rank 0 process,
-            # the `barrier` invocation causes deadlock,
-            # so I implement `_save_checkpoint_rank_zero_only` instead of
-            # `trainer.save_checkpoint`.
-            self._save_checkpoint_rank_zero_only(
-                trainer,
-                tmp_model_save_path,
-            )
-            self.client.log_artifact(self.run_id, tmp_model_save_path, checkpoint_artifact_dir)
-        finally:
-            shutil.rmtree(tmp_dir, ignore_errors=True)
+    def checkpoint_file_suffix(self):
+        return "pth"
```

Comment:
This function is just returning a hard-coded string, can we make it a class attribute? `self._checkpoint_file_suffix=""pth`.